[Joshy837] iP

Joshy837 opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Joshy837 committed:
Rename, greet, exit

Joshy837 committed:
Echo

Joshy837 committed:
Add, List

Joshy837 committed:
Mark as Done

Joshy837 committed:
ToDos, Events, Deadlines

Joshy837 committed:
Automated Text UI Testing

Joshy837 committed:
Handle Errors

Joshy837 committed:
Delete

Joshy837 committed:
Refactored code

Joshy837 committed:
Load & Save

Joshy837 committed:
test branching

Joshy837 committed:
test branch 2

Joshy837 committed:
test branch 3

Joshy837 committed:
removed branch comments

Joshy837 committed:
abstracted code into Action class

Joshy837 committed:
Dates and Times

Joshy837 committed:
Added more OOP

Joshy837 committed:
Deleted old files

Joshy837 committed:
renamed Ben file

Joshy837 committed:
Organised files into different packages

Joshy837 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Joshy837 committed:
added JUnit, added LocalDate to Event Task type

Joshy837 committed:
added JUnit Test cases

Joshy837 committed:
Added Jar support

Joshy837 committed:
added JavaDocs for all classes

Joshy837 committed:
tweaked code to follow the coding standard

Joshy837 committed:
added Find feature

Joshy837 committed:
Merge branch 'branch-A-JavaDoc'

Joshy837 committed:
Fixed merge conflicts

Joshy837 committed:
Fixed additional merge conflicts

zhuuyicheng commented on src/main/java/ben/tasks/TaskList.java:
```diff
@@ -0,0 +1,201 @@
+ */
+public class TaskList {
+
+    private List<Task> tasks;
```
I like that you used plural form on names representing a collection of objects 👍

zhuuyicheng commented on src/main/java/ben/tasks/TaskList.java:
```diff
@@ -0,0 +1,201 @@
+    /**
+     * Gets the string representation of a task at the specified index.
+     *
+     * @param index The index of the task.
+     * @return The string representation of the task.
+     */
+    public String toString(int index) {
+        Task currTask = this.tasks.get(index);
+        return currTask.toString();
+    }
+
+    /**
+     * Displays the entire TaskList.
+     */
+    public void showTaskList() {
+        for (int i = 0; i < this.tasks.size(); i++) {
+            Task currTask = this.tasks.get(i);
+            Ui.show((i + 1) + ". " + currTask);
+        }
+    }
+
+    /**
+     * Adds a task to the TaskList.
+     *
+     * @param task The task to be added.
+     */
+    public void addTask(Task task) {
+        this.tasks.add(task);
+    }
+
+    /**
+     * Removes a task at the specified index from the TaskList.
+     *
+     * @param index The index of the task to be removed.
+     * @return The removed task.
+     */
+    public Task removeTask(int index) {
+        Task deletedTask = this.tasks.remove(index);
+        return deletedTask;
+    }
```
I like that your methods (elsewhere as well) are named as verbs and written in camelCase, following the coding standards

zhuuyicheng commented on src/main/java/ben/tasks/TaskList.java:
```diff
@@ -0,0 +1,201 @@
+     *
+     * @return A string representing the formatted TaskList for saving to a file.
+     */
+    public String formatSave() {
```
It might be possible to rename this method such that it is clearer about what it does? Perhaps something like "convertToSaveFormat" might be a little clearer? Just my personal opinion as I was unsure of what the method did at first

zhuuyicheng commented on src/main/java/ben/tasks/Task.java:
```diff
@@ -0,0 +1,72 @@
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
I like that your boolean variables are named to sound like booleans, as per the coding standards

zhuuyicheng commented on src/main/java/ben/commands/MarkCommand.java:
```diff
@@ -0,0 +1,49 @@
+ * Represents a command to mark a task as done in the task list.
+ */
+public class MarkCommand extends Command {
+    private final int index;
```
I like that your class variables are never declared public which follows the coding standard

STELLA-LYE commented on src/main/java/ben/ui/Ui.java:
```diff
@@ -0,0 +1,157 @@
+     *
+     * @param message The message to be displayed.
+     */
+    public static void show(String message) {
```
Perhaps could be more specific, something like showMessage?

STELLA-LYE replied:
Perhaps the function name can be more detailed like showMessage?

STELLA-LYE commented on src/main/java/ben/ui/Ui.java:
```diff
@@ -0,0 +1,157 @@
+    /**
+     * Displays a message indicating that no matching tasks have been found.
+     */
+    public void showNoTasksFoundMessage() {
```
I like how each function has a specific and clear name. It is concise and very neat.

STELLA-LYE replied:
I like how each function name is detailed and concise.

STELLA-LYE reviewed (commented):
LGTM.  I found your code easy to read and well-written. The function names are detailed and easy understand. There is good naming convention used.

STELLA-LYE commented on src/test/java/ben/tasks/TodoTest.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    @Test
+    public void testToStringCompletedTodo() {
```
Perhaps can consider using shorter function names?

STELLA-LYE commented on src/main/java/ben/storage/Storage.java:
```diff
@@ -0,0 +1,88 @@
+     * @param saveInput The string representation of tasks to be saved.
+     * @throws BenException If there is an issue with the file or folder paths.
+     */
+    public void save(String saveInput) throws BenException {
```
Perhaps can use more detailed naming like saveTask ?

STELLA-LYE reviewed (commented):
LGTM. I found your code easy to read and well-written. I like how all the functions are detailed and concise, making your code easy to understand.

SeaHogs commented on src/main/java/ben/storage/Storage.java:
```diff
@@ -0,0 +1,88 @@
+    private static void writeToFile(String filePath, String textToAdd) throws IOException {
+        try (FileWriter fw = new FileWriter(filePath)) {
+            fw.write(textToAdd);
+        }
```
Maybe having a catch block for try will give more clarity when error is thrown?

SeaHogs commented on src/main/java/ben/commands/EventCommand.java:
```diff
@@ -0,0 +1,48 @@
+ * Represents a command to add a new event task.
+ */
+public class EventCommand extends Command {
+    private final String description;
```
I like that the fields (description, startDate, endDate) are declared as private and final, promoting encapsulation and information hiding.

SeaHogs commented on src/main/java/ben/parser/Parser.java:
```diff
@@ -0,0 +1,154 @@
+                }
+
+                int index = Integer.parseInt(tokens[1]) - 1;
+
```
There is some duplicated code for parsing the index of a task (e.g., in the "mark," "unmark," and "delete" cases). Maybe extracting this logic into a separate method can to avoid redundancy? (if it doesn't affect readability)

SeaHogs commented on src/main/java/ben/commands/DeadlineCommand.java:
```diff
@@ -0,0 +1,45 @@
+        Task newDeadline = new Deadline(false, description, deadline);
+        tasks.addTask(newDeadline);
+
+        ui.showAddedTaskMessage();
```
I like that all statements are at the same level of abstraction.

SeaHogs commented on src/main/java/ben/Main.java:
```diff
@@ -0,0 +1,67 @@
+                String fullCommand = ui.readCommand();
+                ui.showLine(); // show the divider line ("_______")
+                Command c = Parser.parse(fullCommand);
+                c.execute(tasks, ui, storage);
```
Perhaps using a more meaningful name then c even if it is only used in the next line

Joshy837 committed:
Update README.md

Joshy837 committed:
Update README.md

Joshy837 committed:
Add JavaFX GUI feature

Joshy837 committed:
Add checkstyle functionality

Joshy837 committed:
Merge branch 'branch-A-Checkstyle'

Joshy837 committed:
Merge remote-tracking branch 'origin/master'

Joshy837 committed:
Add JavaFX functionality

Joshy837 committed:
Add GUI functionality

Joshy837 committed:
Modify dialog box sizing

Joshy837 committed:
Merge branch 'branch-Level-10'

Joshy837 committed:
Update new Ben.jar file with GUI

Joshy837 committed:
Add assertions in DeleteCommand, MarkCommand and UnmarkCommand

Delete, Mark and Unmark commands have some common field validation checking behaviours.

The field validation can be captured using assertions instead of checked exceptions in production code.

Replacing the validation checks with assertions allows us to check for invalid inputs by the programmer. For example, assertions can be used to check for inputs that are out of bounds of the possible valid integers.

Let's replace all these validation checks with assertions to make the valid cases more explicit.

Using assertions is preferable over if statements and checked exceptions because it makes the validation checks more explicit to the programmer.

Joshy837 committed:
Refactor code in Parser.java

Parse method in Parser.java is very long and contains some common behaviours (e.g. checkEmptyFields())

The common behaviours across the different commands parsed cause code duplication.

Extracting the common behaviour into another function, checkEmptyField(), allows us to perform abstraction when dealin with commands. For instance, validity checking can be performed with only 1 function, instead of having many individual if conditions.

Also, methods for each command are abstracted out into each of their own parser methods, which are called whenever the command is called.

Refactoring these code to ensure abstraction is preferable to having a whole chunk of code as it improves code quality

Joshy837 committed:
Merge pull request #2 from Joshy837/branch-A-CodeQuality

Branch a code quality

Joshy837 committed:
Merge pull request #3 from Joshy837/branch-A-Assertions

Add assertions in DeleteCommand, MarkCommand and UnmarkCommand

Joshy837 committed:
Merge branch 'branch-A-CodeQuality'

Joshy837 committed:
Merge branch 'branch-A-Assertions'

Joshy837 committed:
Merge branch 'master' of https://github.com/Joshy837/ip

Joshy837 committed:
Refactor Ui.java and add varargs for UI display

Joshy837 committed:
Add JavaDocs to Ui.java

Joshy837 committed:
add Findin better search

Joshy837 committed:
Merge branch 'branch-A-BCD-Extension'

Joshy837 committed:
add user guide

Joshy837 committed:
Add Ui.png
